signer/src/storage/postgres.rs

```diff
@@ -1553,6 +1555,152 @@ impl super::DbRead for PgStore {
+                  , wr.block_hash
+                  , wr.recipient
+                  , wr.amount
+                  , wr.max_fee
+                  , wr.sender_address
+                  , wr.bitcoin_block_height
+                FROM sbtc_signer.withdrawal_requests wr
+                -- Request confirmed on stacks chain
+                JOIN stacks_context_window sc ON wr.block_hash = sc.block_hash
+                -- Request not accepted
+                LEFT JOIN sbtc_signer.bitcoin_withdrawals_outputs AS bwo
+                  ON bwo.request_id = wr.request_id
+                 AND bwo.stacks_block_hash = wr.block_hash
+                LEFT JOIN bitcoin_transactions AS bc_trx
+                  ON bc_trx.txid = bwo.bitcoin_txid
+                LEFT JOIN bitcoin_blockchain
+                  ON bc_trx.block_hash = bitcoin_blockchain.block_hash
+                -- Request not rejected
+                LEFT JOIN withdrawal_reject_events AS wre
+                  ON wre.request_id = wr.request_id
+                LEFT JOIN stacks_context_window sc2
+                  ON wre.block_hash = sc2.block_hash
+                WHERE
+                    -- Request is final
+                    wr.bitcoin_block_height <= $4
+
+                -- we need to group since we could have multiple withdrawals 
+                -- outputs for a single request, and some of them may not be in
+                -- the canonical chain, resulting in a NULL bc_trx.block_hash;
+                -- so we group and check that all the rows have NULL
+                GROUP BY
+                    wr.request_id
+                  , wr.txid
+                  , wr.block_hash
+                  , wr.recipient
+                  , wr.amount
+                  , wr.max_fee
+                  , wr.sender_address
+                  , wr.bitcoin_block_height
+                HAVING
+                    -- Request not accepted (cont'd)
+                    COUNT(bc_trx.block_hash) = 0
+                    -- Request not rejected (cont'd)
+                AND COUNT(sc2.block_hash) = 0
+            )
+            -- Now from the above pick the ones having either enough votes
+            -- against, or have expired
+            SELECT
+                wr.request_id
+              , wr.txid
+              , wr.block_hash
+              , wr.recipient
+              , wr.amount
+              , wr.max_fee
+              , wr.sender_address
+              , wr.bitcoin_block_height
+            FROM pending_withdrawals wr
+            JOIN sbtc_signer.withdrawal_signers signers USING(request_id, block_hash)
+            WHERE NOT signers.is_accepted
+            GROUP BY wr.request_id
+                   , wr.txid
+                   , wr.block_hash
+                   , wr.recipient
+                   , wr.amount
+                   , wr.max_fee
+                   , wr.sender_address
+                   , wr.bitcoin_block_height
+            HAVING COUNT(wr.request_id) > $5
```

Comment:
We might be able to simplify all of this. I propose that we only reject withdrawal requests after they have expired.

Reply:
I forgot to mention that I detail why in https://github.com/stacks-network/sbtc/issues/1363. If we only reject withdrawal requests after they have expired, it would allow for some simplification of the query, and what needs to be tested.

The flip side is that I recommend us looking into the mempool to know if a withdrawal request is being serviced by a sweep. If it is then we don't create `reject-withdrawal-request` contract calls for the associated withdrawal request. But the mempool stuff doesn't impact this PR though, it just adds more work 😅 overall.

Reply:
Removed the votes in [bc0cb37](https://github.com/stacks-network/sbtc/pull/1362/commits/bc0cb3782ebad2625bb1344b41c20aac9d34fe4c)